solana/ts/lib/ntt.ts

```diff
@@ -0,0 +1,1150 @@
+    const registeredTransceiver = (transceiver: PublicKey): PublicKey =>
+      derivePda(["registered_transceiver", transceiver.toBytes()], programId);
+    const transceiverMessageAccount = (
+      chain: Chain,
+      id: Uint8Array
+    ): PublicKey =>
+      derivePda(["transceiver_message", chainToBytes(chain), id], programId);
+    const wormholeMessageAccount = (outboxItem: PublicKey): PublicKey =>
+      derivePda(["message", outboxItem.toBytes()], programId);
+    const lutAccount = (): PublicKey => derivePda("lut", programId);
+    const lutAuthority = (): PublicKey => derivePda("lut_authority", programId);
+    const sessionAuthority = (
+      sender: PublicKey,
+      args: TransferArgs
+    ): PublicKey =>
+      derivePda(
+        [
+          "session_authority",
+          sender.toBytes(),
+          keccak256(
+            encoding.bytes.concat(
+              encoding.bytes.zpad(new Uint8Array(args.amount.toArray()), 8),
+              chainToBytes(args.recipientChain.id),
+              new Uint8Array(args.recipientAddress),
+              new Uint8Array([args.shouldQueue ? 1 : 0])
+            )
+          ),
+        ],
+        programId
+      );
+
+    // TODO: memoize?
+    return {
+      configAccount,
+      outboxRateLimitAccount,
+      inboxRateLimitAccount,
+      inboxItemAccount,
+      sessionAuthority,
+      tokenAuthority,
+      emitterAccount,
+      wormholeMessageAccount,
+      peerAccount,
+      transceiverPeerAccount,
+      transceiverMessageAccount,
+      registeredTransceiver,
+      lutAccount,
+      lutAuthority,
+    };
+  };
+
+  export async function getVersion(
+    connection: Connection,
+    programId: PublicKey,
+    sender?: PublicKey
+  ): Promise<IdlVersion> {
+    // the anchor library has a built-in method to read view functions. However,
+    // it requires a signer, which would trigger a wallet prompt on the frontend.
+    // Instead, we manually construct a versioned transaction and call the
+    // simulate function with sigVerify: false below.
+    //
+    // This way, the simulation won't require a signer, but it still requires
+    // the pubkey of an account that has some lamports in it (since the
+    // simulation checks if the account has enough money to pay for the transaction).
+    //
+    // It's a little unfortunate but it's the best we can do.
+
+    if (!sender) {
+      const address =
+        connection.rpcEndpoint === rpc.rpcAddress("Devnet", "Solana")
+          ? "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J" // The CI pubkey, funded on ci network
+          : connection.rpcEndpoint.startsWith("http://localhost")
+          ? "98evdAiWr7ey9MAQzoQQMwFQkTsSR6KkWQuFqKrgwNwb" // the anchor pubkey, funded on local network
+          : "Hk3SdYTJFpawrvRz4qRztuEt2SqoCG7BGj2yJfDJSFbJ"; // The default pubkey is funded on mainnet and devnet we need a funded account to simulate the transaction below
```

Comment:
This seems like a more general pattern/problem that we might want solve at a higher level (so we don't have magic values hardcoded here)

Reply:
```suggestion
        //TODO pull out in a separate util
        connection.rpcEndpoint === rpc.rpcAddress("Devnet", "Solana")
          ? "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J" // The CI pubkey, funded on ci network
          : connection.rpcEndpoint.startsWith("http://localhost")
          ? "98evdAiWr7ey9MAQzoQQMwFQkTsSR6KkWQuFqKrgwNwb" // the anchor pubkey, funded on local network
          : "Hk3SdYTJFpawrvRz4qRztuEt2SqoCG7BGj2yJfDJSFbJ"; // The default pubkey is funded on mainnet and devnet we need a funded account to simulate the transaction below
```

Reply:
created #459 to track this